toolchain/check/check_unit.cpp

```diff
@@ -325,6 +334,34 @@ auto CheckUnit::ImportOtherPackages(SemIR::TypeId namespace_type_id) -> void {
   }
 }
 
+auto CheckUnit::ImportCppPackages() -> void {
+  const auto& imports = unit_and_imports_->cpp_imports;
+  if (imports.empty()) {
+    return;
+  }
+
+  if (imports.size() >= 2) {
+    context_.TODO(imports[1].node_id,
+                  "multiple Cpp imports are not yet supported");
+    return;
+  }
+
+  const auto& import = imports.front();
+  llvm::StringRef filename =
+      unit_and_imports_->unit->value_stores->string_literal_values().Get(
+          import.library_id);
+
+  // TODO: Pass the import location so that diagnostics would point to it.
+  auto source_buffer = SourceBuffer::MakeFromFile(
+      *fs_, filename, unit_and_imports_->err_tracker);
```

Comment:
Could we pass `import.library_id` here to MakeFromFile, and let it default to Invalid, and when Invalid it can use the filename as the location (that's weird to me but :shrug:)?

Reply:
To be sure, I think this call should go away with the `#include` migration. I'd prefer not to change the API if this is temporary; I'd mainly suggested it in order to minimize/consolidate file error handling.

Reply:
Ah okay, nevermind then. We can just leave them broken for now.

Reply:
> Could we pass `import.library_id` here to MakeFromFile, and let it default to Invalid, and when Invalid it can use the filename as the location (that's weird to me but 🤷)?

FWIW, that's pretty much what I had in mind to do as a followup and why I put a TODO.
I agree though that migrating to using `#include` would make this obsolete so we should probably do that first.